CRM/Contactlayout/BAO/ContactLayout.php

```diff
@@ -119,6 +125,65 @@ public static function getBlock($fullName) {
     return NULL;
   }
 
+  /**
+   * Determines if the block can be displayed for the given contact type.
+   *
+   * If the block is for a contact's relation then we determine if the given
+   * contact type and the relation's contact type match.
+   *
+   * When the block has no relation we match the block's contact type to the
+   * given contact type.
+   *
+   * @param array $blockInfo
+   * @param string $blockRelation
+   * @param string $contactType
+   * @return bool
+   */
+  protected static function checkBlockValidity ($blockInfo, $blockRelation = NULL, $contactType = NULL) {
+    if ($blockRelation) {
+      $relationship = self::getRelationshipFromOption($blockRelation);
+
+      return
+        ($relationship['direction'] === 'r' && (
+          ($contactType === $relationship['type']['contact_type_a'] &&
+            $blockInfo['contact_type'] === $relationship['type']['contact_type_b']) ||
+          ($contactType === $relationship['type']['contact_type_b'] &&
+            $blockInfo['contact_type'] === $relationship['type']['contact_type_a']))) ||
+        ($relationship['direction'] === 'ab' && (
+          $blockInfo['contact_type'] === $relationship['type']['contact_type_a'] &&
+          $contactType === $relationship['type']['contact_type_b'])) ||
+        ($relationship['direction'] === 'ba' && (
+          $blockInfo['contact_type'] === $relationship['type']['contact_type_b'] &&
+          $contactType === $relationship['type']['contact_type_a']));
+    } else {
+      return $blockInfo && (!$contactType || empty($blockInfo['contact_type']) || $contactType == $blockInfo['contact_type']);
+    }
+  }
+
+  /**
+   * Returns the relationship type and direction for the given parameter.
+   *
+   * The parameter might come in the format `15_ab` where `15` is the relationship
+   * type ID, and `ab` is the direction.
+   *
+   * @param string $relationshipOption
+   * @return array
+   */
+  protected static function getRelationshipFromOption ($relationshipOption) {
+    $relationship = explode('_', $relationshipOption);
+    $relationshipTypeId = $relationship[0];
+    $relationshipType = \Civi\Api4\RelationshipType::get()
+      ->addWhere('id', '=', $relationshipTypeId)
+      ->addWhere('is_active', '=', TRUE)
```

Comment:
```suggestion
      ->addWhere('id', '=', $relationshipTypeId)
      ->setCheckPermissions(FALSE)
```

I'm recommending that you disable permission checks for this api call (since it is backend data not being shown to the user) and also removing the `is_active` clause as your code doesn't actually have any safeguards in case the relationship type isn't found. In addition to removing the `is_active` clause I'd recommend adding some safeguard if the relationship type has been deleted. Perhaps log the error and return false.